apidocs.yaml

```diff
@@ -0,0 +1,849 @@
+# SPDX-FileCopyrightText: 2024 PNED G.I.E.
+#
+# SPDX-License-Identifier: Apache-2.0
+openapi: 3.0.3
+info:
+  title: Access Request Management Service - OpenAPI 3.0
+  description: |-
+    This is the OpenAPI 3.0 specification for the Access Request Management Service.
+    This service is responsible for managing access requests to the Genomic Data Infrastructure (GDI).
+    The service provides endpoints for creating, updating, and retrieving access requests, as well as managing the datasets and members associated with each request.
+    The service also provides endpoints for managing baskets of access requests.
+
+  #termsOfService: http://TODO/terms/
+  contact:
+    email: gdi-WP4@elixir-europe.org
+  license:
+    name: Apache 2.0
+    url: https://spdx.org/licenses/Apache-2.0.html
+  version: 0.0.0
+externalDocs:
+  description: Find out more about GDI
+  url: https://genomicdatainfrastructure.github.io/gdi-userportal-docs/
+servers:
+  - url: https://localhost:8080
+    description: Development server
+tags:
+  - name: applications
+    description: All endpoints related to access request management
+  - name: baskets
+    description: All endpoints related to baskets
+  - name: attachments
+    description: All endpoints related to attachments
+  - name: members
+    description: All endpoints related to members
+  - name: terms
+    description: All endpoints related to terms
+  - name: datasets
+    description: All endpoints related to datasets
+  - name: forms
+    description: All endpoints related to forms
+  - name: actions
+    description: All endpoints related to user actions
+  - name: events
+    description: All endpoints related to events
+paths:
+  /api/v1/applications:
+    get:
+      summary: List applications
+      operationId: list_applications_v1
+      tags:
+        - "applications"
+      responses:
+        "200":
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                type: array
+                title: collection of applications
+                items:
+                  $ref: "#/components/schemas/ListedApplication"
+      security:
+        - auth:
+          - read:applications
+  /api/v1/applications/{id}:
+    get:
+      summary: Retrieve application
+      operationId: retrieve_application_v1
+      tags:
+        - "applications"
+      parameters:
+        - name: id
+          in: path
+          description: ID of application to retrieve
+          required: true
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Successful Response
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/RetrievedApplication"
+      security:
+        - auth:
+          - read:applications
+  /api/v1/applications/{id}/save-forms-and-duos:
```

Comment:
what is duo here?

Reply:
Thanks for spotting it. It's fixed. It's another part of the application that is saved at the same moment the forms are saved.